src/lib/profiles/data-management/Current/NotificationEngine.cpp

```diff
@@ -1680,13 +1680,15 @@ WEAVE_ERROR NotificationEngine::BuildSubscriptionlessNotification(PacketBuffer *
         TraitDataHandle traitHandle = currPath->mTraitDataHandle;
 
         // Get the max version from the datasource
-        err = pubCatalog->Locate(traitHandle, &dataSource);
-        SuccessOrExit(err);
-
-        schemaVersion = dataSource->GetSchemaEngine()->GetMaxVersion();
+        // Locate() can return an error if the sink has been removed from the catalog. In that case,
+        // skip this path
+        if (pubCatalog->Locate(traitHandle, &dataSource) == WEAVE_NO_ERROR)
```

Comment:
I would prefer that this code not ignore all errors, only the error indicating that the data source wasn't found.